src/main/java/mate/dao/impl/DriverDaoImpl.java

```diff
@@ -109,4 +112,32 @@ private Driver getDriver(ResultSet resultSet) throws SQLException {
         driver.setId(id);
         return driver;
     }
+
+    @Override
+    public Optional<Driver> findByLogin(String login) {
+        String query = "SELECT * FROM drivers WHERE login = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            Driver driver = null;
+            if (resultSet.next()) {
+                driver = getDriverByLogin(resultSet);
```

Comment:
can't we use our private getDriver(ResultSet resultSet) method here?

Reply:
S.O.L.I.D : O - open-closed principe. I expanding this class(i created this and below methods) and didn`t modificate it.